index.rst

```diff
@@ -42,20 +42,151 @@
+
+.. [*] In :cite:`LTS-836` the group ID is called a "transaction ID" since it is associated with a single queue transaction.
+
+Processing Data as it Arrives
+=============================
+
+All triggered processing should be done through the "Gen3" middleware pipeline system consisting of a pipeline of``PipelineTask``.
+Each trigger will need to specify at minimum the name of the Pipeline to execute and the data ID that should be processed.
+Additionally it may be necessary to provide an explicit config file by name.
+Using standard change-controlled pipeline configurations is preferable to scripts trying to send override configuration on-the-fly since it provides a self-contained way to reproduce the processing.
+
+For a Pipeline to be executed the data ID needs to be converted to a quantum graph suitable for passing to the activator.
+The data ID would usually be the obsid for the data just acquired.
+Furthermore the group ID could be used to obtain all data that have previously been taken related to this observation.
 
 .. note::
 
-   **This technote is not yet published.**
+  No LSST rapid analysis pipelines are required to co-add data incrementally.
+  Pipelines should not co-add data as it arrives and should always defer co-addition until the final observation in the group has been taken.
+
+Single frame processing is the simplest mode whereby the products from the processing are not to be combined with subsequent observations.
+The more complex scenario relates to processing that requires co-addition where the co-addition can only occur at the end of data acquisition.
+
+One proposal for this is to have a chained pipeline approach.
+As data arrives they are processed with the nominated single frame processing variant.
+On completion they register themselves as inputs to a combination pipeline.
+This pipeline does not run automatically but waits to be told that all data have arrived.
+If an observation fails to reduce properly, or fails to pass some metrics threshold it will not register with the combination pipeline.
+
+When all expected data have arrived and been processed on their own, the final combination pipeline will be unblocked and will run to completion.
+Events will be issued on the completion of a single frame processing and on the completion of the combination pipeline.
+Metrics will also be published where appropriate during processing although the mechanism for this has not been defined.
+
+The issues that have to be solved are:
+
+1. How does the pipeline know to start processing a data ID?
+2. Which pipeline should be run for the given data ID?
+3. If the pipeline supports chaining, is that known to the system passing in the data?
+4. Who decides which observations should be combined and how?
+5. How does the system know the final observation arrived such that the combination pipeline can run?
+
+This document will describe two alternate scenarios for answering the above questions.
```

Comment:
Note that these scenarios are not mutually exclusive.  We can have both, if need be.

Reply:
Yes, we can have the complexity in multiple places and have different states in the OCPS.